buddy_list: Show unsubscribed participants in participants list.

evykassirer opened the pull request:
Followup to https://github.com/zulip/zulip/pull/32470#issuecomment-2504958549

timabbott commented:
Please add comments documenting the reasoning here, and ideally test coverage if that's easy (I expect it should be?) -- if there's a subtle detail like this that can be broken, it can be broken in refactoring too.

evykassirer commented:
It's not the most straight forward to add a test. We don't actually even have any test coverage for participants in the buddy_list test file. To more easily test, we'd probably want to split the logic out more from the UI/JQuery code. I can also spend time setting up the necessary mocks and stuff to add more test coverage as is, if you want!

I've updated the commit to add the motivation, and also fixed a linter failure.

timabbott commented:
Merged, thanks @evykassirer!
